pvanalytics/features/daytime.py

```diff
@@ -0,0 +1,247 @@
+"""Functions for identifying daytime"""
+import numpy as np
+import pandas as pd
+from pandas.tseries import frequencies
+
+
+def _rolling_by_minute(data, days, f):
+    # apply `f` to a rolling window of length `days` at each minute of
+    # the day.
+    rolling = data.groupby(
+        data.index.hour * 60 + data.index.minute
+    ).rolling(
+        min_periods=1,
+        center=True,
+        window=days
+    )
+    result = f(rolling).reset_index(0, drop=True)
+    return result.sort_index()
+
+
+def _run_lengths(series):
+    # Count the number of equal values adjacent to each value.
+    #
+    # Examples
+    # --------
+    # >>> _run_lengths(pd.Series([True, True, True]))
+    # 0    3
+    # 1    3
+    # 2    3
+    #
+    # >>> _run_lengths(
+    # ...     pd.Series([True, False, False, True, True, False]
+    # ... ))
+    # 0    1
+    # 1    2
+    # 2    2
+    # 3    2
+    # 4    2
+    # 5    1
+    runs = (series != series.shift(1)).cumsum()
+    return runs.groupby(runs).transform('count')
+
+
+def _to_numeric(series):
+    # Convert a Boolean series to a numeric series
+    #
+    # False -> 1, True -> 2, NA -> 3
+    numeric_series = pd.Series(index=series.index, dtype='float64')
+    numeric_series.loc[~series] = 1
+    numeric_series.loc[series] = 2
+    numeric_series.loc[series.isna()] = 3
+    return numeric_series
+
+
+def _from_numeric(series):
+    # Inverse of :py:func:`_to_numeric()`
+    boolean_series = series == 2
+    boolean_series.loc[series == 3] = np.nan
+    return boolean_series.astype('bool')
+
+
+def _smooth_if_invalid(series, invalid, correction_window):
+    # replace values in `series` marked as True in `invalid` with the
+    # `correction_window`-day rolling median at each minute.
```

Comment:
seems like a different term than 'median' would help, to avoid confusing this step with the 'low median' criteria. What is being done (if I follow correctly) is a voting algorithm, where a minute in an invalid window is relabeled to be the same as the majority label of minutes in the correction_window